connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsIntegrationBaseTest.java

```diff
@@ -319,27 +319,21 @@ public void testReplication() throws Exception {
         waitForCondition(() -> primaryClient.remoteConsumerOffsets(consumerGroupName, BACKUP_CLUSTER_ALIAS,
             Duration.ofMillis(CHECKPOINT_DURATION_MS)).containsKey(new TopicPartition("backup.test-topic-1", 0)), CHECKPOINT_DURATION_MS, "Offsets not translated downstream to primary cluster.");
 
-        waitForCondition(() -> primaryClient.remoteConsumerOffsets(consumerGroupName, BACKUP_CLUSTER_ALIAS,
-            Duration.ofMillis(CHECKPOINT_DURATION_MS)).containsKey(new TopicPartition("test-topic-1", 0)), CHECKPOINT_DURATION_MS, "Offsets not translated upstream to primary cluster.");
```

Comment:
And why should we remove this wait condition? Please help me understand it. Thanks.

Reply:
This was querying checkpoints from the `backup.checkpoints.internal` topic. This topic is on the primary cluster and contains data emitted by the backup->primary checkpoint connector. 

In the backup->primary mirroring flow, the remote topics (that end up in the `primary` cluster) are all prefixed with `backup.`. With this change we now only emit checkpoints for topics present in offset-syncs, so only topics prefixed with `backup.`.

So the assertion above still works, the checkpoint connector has emitted checkpoints for `backup.test-topic-1`. But this assertion is not true anymore for `test-topic-1`. This makes sense because in the primary cluster, `test-topic-1` is a local topic, so the backup->primary checkpoint connector has no business committing offsets for that topic.

Reply:
For this types of changes, a diagram often helps.

We see that `mm2-offset-syncs.primary.internal` only contains mapping from `test-topic-1` on the backup cluster to `backup.test-topic-1` on the primary cluster. Then when backup->primary checkpoint runs it's only able to compute committed offsets for `backup.test-topic-1`, it has no information about offsets for the local `test-topic-1` in the primary cluster.
<img width="1230" alt="image" src="https://user-images.githubusercontent.com/903615/168580961-8f3e5860-cae3-43bf-9d36-ecd7553e42eb.png">

Reply:
Thanks for the clear explanation. I understand now.